### PR TITLE
Receiving sms

### LINKED ISSUE #3: 4.1.3 Receiving SMS
Feature 4.1.3
- Main scenario

Programmer: 
Tester: 
Code review:

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: UE/Application/Sms.cpp [hunk 1]
```diff
@@ -8,6 +8,7 @@ Sms::Sms(common::PhoneNumber from, std::string text)
     : from(from),
       text(text)
 {
+    this->read = false;
```
Comment (jakjare) [inline UE/Application/Sms.cpp]: `this` is not necessary here.
```suggestion
    read = false;
```

**Review by jakjare — CHANGES_REQUESTED** [review]
Your tests on branch `receiving-sms` failed:

> [  FAILED  ] 3 tests, listed below:
[  FAILED  ] SmsDbTestSuite.shallAddUnreadSms
[  FAILED  ] SmsDbTestSuite.shallGetSms
[  FAILED  ] UserPortTestSuite.shallShowMenuOnConnected
 3 FAILED TESTS

**Review by jakjare — APPROVED** [review]

File: UE/Application/Application.hpp [hunk 2]
```diff
@@ -31,6 +31,7 @@ class Application : public IEventsHandler
     void handleSib(common::BtsId btsId) override;
     void handleAttachAccept() override;
     void handleAttachReject() override;
+    void handleSmsReceived(common::PhoneNumber from, std::string text) override;
```
Comment (Koci0) [inline UE/Application/Application.hpp]: Use reference passing for the `std::string`

File: UE/Application/Ports/IBtsPort.hpp [hunk 3]
```diff
@@ -14,6 +15,7 @@ class IBtsEventsHandler
     virtual void handleSib(common::BtsId) = 0;
     virtual void handleAttachAccept() = 0;
     virtual void handleAttachReject() = 0;
+    virtual void handleSmsReceived(common::PhoneNumber, std::string) = 0;
```
Comment (Koci0) [inline UE/Application/Ports/IBtsPort.hpp]: Use reference passing for the `std::string`.

File: UE/Application/Ports/UserPort.cpp [hunk 4]
```diff
@@ -61,11 +61,10 @@ void UserPort::showSmsList() {
         for(auto sms : ListSms) {
             menu.addSelectionListItem("From: " + to_string(sms.from), sms.text);
         }
+        gui.setAcceptCallback([&](){
+            showSms(menu.getCurrentItemIndex().second);;
```
Comment (Koci0) [inline UE/Application/Ports/UserPort.cpp]: Double semicolon.
```suggestion
            showSms(menu.getCurrentItemIndex().second);
```

File: UE/Application/SmsDb.cpp [hunk 5]
```diff
@@ -5,15 +5,7 @@
...
-{
-    std::vector<Sms> vec;
-    // TODO: remove debug vector
-    //init vector for debug purpose
-    std::string s1 = "Alamakota";
-    this->v_sms = vec;
-    const common::PhoneNumber number{11};
-    v_sms.push_back(Sms(number, s1));
-}
+{}
```
Comment (Koci0) [inline UE/Application/SmsDb.cpp]: If we aren't initializing `v_sms` vector here (which I find strange), then maybe we could use `= default` in the header?

File: UE/Application/States/BaseState.hpp [hunk 6]
```diff
@@ -21,6 +21,7 @@ class BaseState : public IEventsHandler
     void handleSib(common::BtsId btsId) override;
     void handleAttachAccept() override;
     void handleAttachReject() override;
+    void handleSmsReceived(common::PhoneNumber from, std::string text) override;
```
Comment (Koci0) [inline UE/Application/States/BaseState.hpp]: Use reference passing for the `std::string`.

File: UE/Application/States/ConnectedState.hpp [hunk 7]
```diff
@@ -13,6 +13,7 @@ class ConnectedState : public BaseState
     // IBtsEventsHandler interface
 public:
     void handleDisconnected() final;
+    void handleSmsReceived(common::PhoneNumber from, std::string text) final;
```
Comment (Koci0) [inline UE/Application/States/ConnectedState.hpp]: Again, use reference for string.

**Review by Koci0 — CHANGES_REQUESTED** [review]

**Review by jakjare — APPROVED** [review]

**Review by Koci0 — APPROVED** [review]

**Review by jakjare — APPROVED** [review]

**Review by robertKozik — APPROVED** [review]

**Pull request merged**